pkg/provisioner/resourcehandlers.go

```diff
@@ -218,6 +218,28 @@ func releaseSecret(sec *corev1.Secret, c kubernetes.Interface) (err error) {
 	return nil
 }
 
+// Remove the finalizer allowing the OBC to finally be deleted.
+func releaseOBC(obc *v1alpha1.ObjectBucketClaim, c versioned.Interface) (err error) {
+	if obc == nil {
+		logD.Info("got nil obc, skipping")
+		return nil
+	}
+	obcNsName := obc.Namespace + "/" + obc.Name
+	obc, err = c.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(obc.Name, metav1.GetOptions{})
```

Comment:
It's not bad to have, but I think it's very unexpected that we will race here, and then the update will just fail and we should retry to delete right? If so then I think it's really safe to remove.

Reply:
Retrying may not be the answer depending on how `Update` is implemented. I've seen Update failures resulting from stale objects. IOW, another thread has updated our copy of the obc. In this case we'd have to do a Get before retrying the Update -- I think...

Reply:
I meant retrying the entire reconcile in case of race.

Reply:
I.m pretty sure that's what naturally occurs when we return an error to `syncHandler` (L153).

Reply:
So that will do the right thing and retry reconcile in case of update race. This is better than adding another `Get()` call just for that negligible race case, and the kube api protects us from that using the generation.

Reply:
I'm tending to agree. As the code is written we always pay the price of a Get for the small chance of a race. Removing the Get is better performing for most cases.  
That said, the pattern of calling Get before adding/removing a finalizer is done through the lib. For better separation of pr focus I'd prefer to see all Gets removed in a separate pr.